Added device tests.

eoneill-mide opened the pull request.

eoneill-mide committed:
Added device tests.

pscheidler commented on test_hardware/conftest.py:
```diff
@@ -0,0 +1,59 @@
+    """
+
+    parser.addoption(
+        "-D", "--device", default=None, help="Specify the serial number of the device to test"
```
```suggestion
        "-D", "--device", default=None, help="Specify the serial number of the device to test, including the initial letter and leading 0s (ie S0016244)"
```

pscheidler replied:
Ignore this, I misinterpretted the situation

pscheidler commented on test_hardware/conftest.py:
```diff
@@ -0,0 +1,59 @@
+    """
+    if config.getoption("--device") is None:
+        pass
+    elif config.getoption("--device")[0] == "S":
```
```suggestion
    elif config.getoption("--device")[0].upper() == "S":
```

pscheidler commented on test_hardware/conftest.py:
```diff
@@ -0,0 +1,59 @@
+        for item in items:
+            if "device_w" in item.keywords:
+                item.add_marker(skip_test)
+    elif config.getoption("--device")[0] == "W":
```
```suggestion
    elif config.getoption("--device")[0].upper() == "W":
```

pscheidler commented on test_hardware/conftest.py:
```diff
@@ -0,0 +1,59 @@
+        pass
+    elif config.getoption("--device")[0] == "S":
+        skip_test = pytest.mark.skip(
+            reason="Run using local option, skipping device required tests")
```
```suggestion
            reason="Test not required for S device")
```

pscheidler commented on test_hardware/conftest.py:
```diff
@@ -0,0 +1,59 @@
+                item.add_marker(skip_test)
+    elif config.getoption("--device")[0] == "W":
+        skip_test = pytest.mark.skip(
+            reason="Run using local option, skipping device required tests")
```
```suggestion
            reason="Test not required for W device")
```

pscheidler reviewed (commented):
Some relatively minor comments, 2 core ideas:
- The --device option should enter the device Type (ie S3-E100D40), not serial number (ie S0014663)
- Print messages and descriptions are important :)

StokesMIDE reviewed (changes requested):
The standard tests already exercise `getDevices()` on the fake recorders; doing it again in a hardware test seems redundant. If it is necessary, can't you use the same copy of the `fake_recorders` as the standard tests?

Commit:
Added Otii files; Minor changes to test_device.

eoneill-mide committed:
Initial draft of setupTeardown() for testing

Commit:
Fixed fake_recorders import

Commit:
Additional fixes to fake_recorders import

eoneill-mide committed:
Initial draft of tests for W specific commands.

eoneill-mide committed:
Document clean up.

eoneill-mide committed:
Merge branch 'ES-842-test-w-specific-commands' into tester/ES-814_device_tests

Commit:
Minor tweaks made to otii files.

eoneill-mide committed:
Added 'requirements' and 'README' files.

eoneill-mide committed:
Added 'device-needed' pytest marker.

StokesMIDE commented:
Are the files in `RawOtiiRecs` needed, or are they an artifact from the EndaqEngTools stuff? I don't see them referenced in the code.

eoneill-mide committed:
Clean up; Minor changes to W tests; Draft of auto test workflow

Commit:
Added GPIO controls to test setup/teardown.

eoneill-mide committed:
GPIO now optional if not on RasPi.

Commit:
Cleaning up recent changes.

Commit:
Added pytest-random-order to requirements.txt

eoneill-mide committed:
Modified tests to work when multiple enDAQs are connected but only one is being tested.

eoneill-mide committed:
Minor fixes to workflow file.

eoneill-mide committed:
Unsupported tests now fail on FW Version <= 2.01

eoneill-mide committed:
Added 'self-hosted' to runs-on in the workflow yml file

eoneill-mide committed:
Adapted tests for old FW (will only work on Raspi)

eoneill-mide committed:
Instances of stopRecOldFW now have is_raspi arg

eoneill-mide committed:
Updated README

eoneill-mide committed:
Added pytest-json-report to requirements

pscheidler committed:
Trying this with the latest test HW

pscheidler committed:
Removing config from device before testing

pscheidler committed:
Adding code to configure the device

pscheidler committed:
Fixing test issues

pscheidler committed:
Fixing test issues

pscheidler committed:
Big changes to delays and how we connect to devices

pscheidler committed:
looking into why the wifi config isn't working and why we can't connect to devices

pscheidler committed:
Lots of changes to tester HW interface, seems to work well on windows

pscheidler committed:
Cutting out most tests to focus on a few odd cases

pscheidler committed:
fixing issues configuring device

pscheidler committed:
adding a sleep to make sure the new config gets to the device

pscheidler committed:
Testing config only

mide-bot committed:
Problems may be from Python 3.13, trying Python 12

pscheidler committed:
separating out venv paths

pscheidler committed:
Adding sleep after start/stop commands

pscheidler committed:
Merged in dev branch

pscheidler committed:
Adding example of config issue

pscheidler committed:
Running into weird import issues

pscheidler committed:
Working on applyConfig fix

pscheidler committed:
Adding a flush to the config write

pscheidler committed:
Tests should work now that device mount is set to flush

pscheidler committed:
Using more get_status, enabling all tests

pscheidler committed:
Increased timeout for W tests, small fixes

pscheidler committed:
Merge branch 'develop' into tester/ES-814_device_tests

pscheidler committed:
Updated venv to be used in each step, added raspi_endaq_controller

JSMTHWCK committed:
added a hello world yaml file to debug local runners

JSMTHWCK committed:
abstracted functions and added new test. Minor adjustments to yml files, removed hello world

JSMTHWCK committed:
fixed bug in helper_functions/general_config.py, added a revert back to old config in setupTeardown

JSMTHWCK committed:
refactored logging, added new tests

JSMTHWCK committed:
Update unit-tests.yml

JSMTHWCK committed:
Update unit-tests.yml

JSMTHWCK committed:
Update unit-tests.yml

JSMTHWCK committed:
Update unit-tests.yml

JSMTHWCK committed:
refactoring, added package for cli testing, fixed bug in general_config.py

JSMTHWCK committed:
Merge branch 'tester/ES-814_device_tests' of github.com:MideTechnology/endaq-device into tester/ES-814_device_tests

JSMTHWCK committed:
refactored, added in new tests; semi-stable state

JSMTHWCK committed:
unified fixture naming convention, added skeletons to test_device.py; unstable commit

JSMTHWCK committed:
added missing imports, removed unecessary tests. Refactored General Config to use hex and match attribute names. Unstable State

JSMTHWCK committed:
various minor changes to make tests work. Semi-stable state

JSMTHWCK committed:
renamed hardware interfaces to match standards. Created device manager class, refactored tests / configs to support it. Semi stable state

JSMTHWCK committed:
created flag --no_tty to skip human-input required tests

JSMTHWCK committed:
removed unecessary files and imports

JSMTHWCK committed:
Finished await tests, removed unecessary fixtures / imports with device_manager update

JSMTHWCK committed:
began README overhaul, w device test start refactoring

JSMTHWCK committed:
fixed bug preventing code from running, skeletons added for new code

JSMTHWCK committed:
renamed device_manager to session_manager; moved tests into seperate files; added more skeletons

JSMTHWCK committed:
forgot to add the test files. Refer to previous commit for changes

JSMTHWCK committed:
changed test collection method for wifi device prep, removed pip requirements and implemented some to compensate

JSMTHWCK committed:
Merge branch 'develop' of github.com:MideTechnology/endaq-device into tester/ES-814_device_tests

JSMTHWCK committed:
changed how storing config information works

JSMTHWCK committed:
Es 814 wifi device tests (#160)

* created new branch for tester wifi capabilities. not tested / working
* reworked how config is stored, unstable state
* renamed marks, disabled ability to use both serial / wifi tests without tty; unstable
* created connection manager to switch connection types, code changed to accommodate

(going ahead with merge w/o review approval; this was reviewed in person and only merging into another `feature` branch with its own PR. - DRS)
